### PR TITLE
Features/http proxy

### PR DESCRIPTION
**Pull request by igorescobar**
Addresses #8 

```elixir
ConfigCat.start_link(sdk_key, http_proxy: "https://myproxy"))
```

With user authorization:

```elixir
ConfigCat.start_link(sdk_key, http_proxy: "https://user:pass@myproxy"))
```

### CODE REVIEW

File: test/config_cat_test.exs [hunk 1]
```diff
@@ -36,7 +36,8 @@ defmodule ConfigCatTest do
       {:ok, client} = start_config_cat(sdk_key, fetch_policy: FetchPolicy.manual())
 
       APIMock
-      |> stub(:get, fn ^url, _headers ->
+      |> stub(:get, fn ^url, _headers, options ->
+        assert options == []
```
Comment (randycoulman) [inline test/config_cat_test.exs]: #minor This can be stated more concisely as `stub(:get, fn ^url, _headers, [] -> `. I normally don't like to put `assert`s in `stub`s, but was forced to do it when testing the headers.

File: test/config_cat_test.exs [hunk 2]
```diff
@@ -144,12 +145,27 @@ defmodule ConfigCatTest do
...
     end
+
+    test "sends proper http proxy options" do
+      {:ok, client} = start_config_cat("SDK_KEY", fetch_policy: FetchPolicy.manual(), http_proxy: "https://myproxy.com")
+
+      response = %Response{status_code: 200, body: %{}}
+
+      APIMock
+      |> stub(:get, fn _url, _headers, options ->
+        assert options == [proxy: "https://myproxy.com"]
```
Comment (randycoulman) [inline test/config_cat_test.exs]: Again, this can be more concisely written as `stub(:get, fn _url, _headers, [proxy: "https://myproxy.com"]`.

When there is an `assert` in a stub, if the code under test is wrong, the stub might not ever get called, and so the assertion would never fire.  As I said above, I was forced to do this for testing the headers, but I try to avoid that pattern as much as possible.

File: test/config_cat_test.exs [hunk 3]
```diff
@@ -250,13 +266,18 @@ defmodule ConfigCatTest do
...
-      |> expect(:get, 1, fn _url, _headers ->
+      |> expect(:get, 1, fn _url, _headers, _options ->
         {:ok, %Response{status_code: 200, body: config}}
       end)
 
-      ConfigCat.get_value(feature, "default", client: client)
-      assert ConfigCat.get_value(feature, "default", client: client) == value
+      result = Enum.reduce 1..call_times, 0, fn _n, _a ->
+        ConfigCat.get_value(feature, "default", client: client)
+      end
```
Comment (randycoulman) [inline test/config_cat_test.exs:275]: What is this change about? Doesn't seem related to the proxy option, and I'm curious about what might be different between calling the function once vs 10 times.

File: test/config_cat_test.exs [hunk 4]
```diff
@@ -270,13 +291,18 @@ defmodule ConfigCatTest do
...
+
+      assert result == value
```
Comment (randycoulman) [inline test/config_cat_test.exs:302]: Here as well. This change seems unrelated to the proxy change.

**Review by randycoulman — APPROVED** [review]
This change looks good!  A couple of minor suggestions and some questions about some test changes.

One concern: does httpoison actually support the `"https://user:pass@myproxy"` style of URL? The docs say that it takes a separate `proxy_auth` key with the username and password. Do we need to split the auth out of the URL and pass it separately, or does HTTPoison do that internally and it's just not documented?

File: test/config_cat_test.exs (hunk 2 above)
Comment (igorescobar) [reply test/config_cat_test.exs]: Good catch! Done! 👍

File: test/config_cat_test.exs (hunk 1 above)
Comment (igorescobar) [reply test/config_cat_test.exs]: Done 👍

File: test/config_cat_test.exs (hunk 3 above)
Comment (igorescobar) [reply test/config_cat_test.exs:275]: It's unrelated. It was just a refactoring, something I stumbled that I thought it was weird. The code was like this:
```
ConfigCat.get_value(feature, "default", client: client)
ConfigCat.get_value(feature, "default", client: client)
```

At first, I thought it was a mistake. Then I removed one line and the tests failed... then I understood what it was about and I tried to make it more explicit of what it is about.

Calling 1 or 10 times, in theory, doesn't make  any difference because the call is cached but I changed back to 2 👍

**Review by randycoulman — APPROVED** [review]
Changes LGTM!
